packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1257,8 +1319,25 @@ export class TransactionController extends BaseController<
         delete txParams.gasPrice;
       }
 
+      if (!this.beforePublish(transactionMeta)) {
+        transactionMeta.status = TransactionStatus.approved;
+        this.updateTransaction(
+          transactionMeta,
+          'TransactionController#approveTransaction - Transaction approved',
+        );
+      }
+
       const unsignedEthTx = this.prepareUnsignedEthTx(txParams);
-      const signedTx = await this.sign(unsignedEthTx, from);
+      const signedTx = await this.sign(
+        unsignedEthTx,
+        from,
+        ...this.getAdditionalSignArguments(transactionMeta),
+      );
+      // In certain cases, the responsibility to publish the transaction lies with the custodian
+      if (!this.afterSign(transactionMeta, signedTx)) {
+        return;
```

Comment:
This `return` here means the signing status isn't changed but also that publish is skipped but that should only be controlled by `beforePublish`.

I think this confirms the need for a private `signTransaction` method as mentioned in another comment.

Reply:
Done https://github.com/MetaMask/core/pull/1787/commits/d59dc3c88ac4c9acce48182e40a4e772bf686fcd